src/Scripting/CSharpTest/ScriptTests.cs

```diff
@@ -313,5 +315,289 @@ public void TestCreateMethodDelegate()
+            var result = await script.EvaluateAsync();
+            Assert.Null(result);
+
+            script = CSharpScript.Create(@"
+bool condition = true;
+if (condition)
+{
+    return 1;
+}
+System.Console.WriteLine();");
+            result = await script.EvaluateAsync();
+            Assert.Equal(1, result);
+        }
+
+        [Fact]
+        public async Task ReturnInNestedScopeWithTrailingVoidExpressionAsInt()
+        {
+            var script = CSharpScript.Create<int>(@"
+bool condition = false;
+if (condition)
+{
+    return 1;
+}
+System.Console.WriteLine();");
+            var result = await script.EvaluateAsync();
+            Assert.Equal(0, result);
+
+            script = CSharpScript.Create<int>(@"
+bool condition = false;
+if (condition)
+{
+    return 1;
+}
+System.Console.WriteLine()");
+            result = await script.EvaluateAsync();
+            Assert.Equal(0, result);
+        }
+
+        [Fact]
+        public async Task ReturnIntWithTrailingDoubleExpression()
+        {
+            var script = CSharpScript.Create(@"
+bool condition = false;
+if (condition)
+{
+    return 1;
+}
+1.1");
+            var result = await script.EvaluateAsync();
+            Assert.Equal(1.1, result);
+
+            script = CSharpScript.Create(@"
+bool condition = true;
+if (condition)
+{
+    return 1;
+}
+1.1");
+            result = await script.EvaluateAsync();
+            Assert.Equal(1, result);
+        }
+
+        [Fact]
+        public async Task ReturnGenericAsInterface()
+        {
+            var script = CSharpScript.Create<IEnumerable<int>>(@"
+if (false)
+{
+    return new System.Collections.Generic.List<int> { 1, 2, 3 };
+}");
+            var result = await script.EvaluateAsync();
+            Assert.Null(result);
+
+            script = CSharpScript.Create<IEnumerable<int>>(@"
+if (true)
+{
+    return new System.Collections.Generic.List<int> { 1, 2, 3 };
+}");
+            result = await script.EvaluateAsync();
+            Assert.Equal(new List<int> { 1, 2, 3 }, result);
+        }
+
+        [Fact]
+        public async Task ReturnNullable()
+        {
+            var script = CSharpScript.Create<int?>(@"
+if (false)
+{
+    return 42;
+}");
+            var result = await script.EvaluateAsync();
+            Assert.False(result.HasValue);
+
+            script = CSharpScript.Create<int?>(@"
+if (true)
+{
+    return 42;
+}");
+            result = await script.EvaluateAsync();
+            Assert.Equal(42, result);
+        }
+
+        [Fact]
+        public async Task ReturnInLoadedFile()
+        {
+            var resolver = TestSourceReferenceResolver.Create(
+                KeyValuePair.Create("a.csx", "return 42;"));
+            var options = ScriptOptions.Default.WithSourceResolver(resolver);       
+
+            var script = CSharpScript.Create("#load \"a.csx\"", options);
+            var result = await script.EvaluateAsync();
+            Assert.Equal(42, result);
```

Comment:
Speaking with @tmat, we feel this behavior is actually incorrect.  I will add an additional commit to prevent trailing expressions in ```#load```ed files from influencing the return value of the script.